File: annni/annni (1).f90
program annni

    implicit none

    real, dimension(2, 2) :: identity 
    complex, dimension(2, 2) :: spin_x, spin_y, spin_z
    real, allocatable, dimension(:, :) :: interaction_one, interaction_two, interaction_three
    integer :: i, j, n, m

    ! Identity

    do i = 1, 2
        do j = 1, 2
            if (i .eq. j) then
                identity(i, j) = 1
            else
                identity(i, j) = 0
            endif
        enddo
    enddo

    ! Spin X

    spin_x(1, 1) = (0, 0)
    spin_x(1, 2) = (1, 0)
    spin_x(2, 1) = (1, 0)
    spin_x(2, 2) = (0, 0)

    ! Spin Y

    spin_y(1, 1) = (0, 0)
    spin_y(1, 2) = (0, -1)
    spin_y(2, 1) = (0, 1)
    spin_y(2, 2) = (0, 0)

    ! Spin Z

    spin_z(1, 1) = (1, 0)
    spin_z(1, 2) = (0, 0)
    spin_z(2, 1) = (0, 0)
    spin_z(2, 2) = (-1, 0)

    n = 4
    m = 4

    allocate (interaction_one(n, m))
    allocate (interaction_two(n, m))
    allocate (interaction_three(n, m))

    interaction_one = 1
    interaction_two = 1
    interaction_three = 1

    ! First interaction
    
    i = 1
    do while (i .lt. n)
        do j = 1, (m - 1)
            interaction_one(i, j) = 2
            interaction_one(i, (j + 1)) = 2

            i = i + 1
        enddo
    enddo

    i = 1
    do while (i .lt. n)
        do j = 1, (m - 2)
            interaction_two(i, j) = 2
            interaction_two(i, (j + 2)) = 2

            i = i + 1
        enddo
    enddo

    i = 1
    do while (i .eq. n)
        do j = 1, m
            interaction_three(i, j) = 2
        enddo
    enddo

    do i = 1, n
        do j = 1, m
            write(*, *) interaction_one(i, j)
            write(*, *) interaction_two(i, j)
            write(*, *) interaction_three(i, j)
        enddo
    enddo

end program

File: annni/annni (1).py
import numpy as np

n, m = 4, 4

identity = np.identity(2)

spin_x = np.array([[0, 1], [1, 0]])
spin_y = np.array([[0, -1j], [1j, 0]])
spin_z = np.array([[1, 0], [0, -1]])

matrix_one = np.ones((n, m))
matrix_two = np.ones((n, m))
matrix_three = np.ones((n, m))

# First interaction
i = 0
for j in range(0, m):
    if j == (m - 1):
        break
    else:
        matrix_one[i][j] = 2
        matrix_one[i][j + 1] = 2

    matrix_one[n - 1][0] = 2
    matrix_one[n - 1][3] = 2

    i = i + 1
    if i >= n:
        break

# Second interaction
i = 0
for j in range(0, m):
    if j == (m - 2):
        break
    else:
        matrix_two[i][j] = 2
        matrix_two[i][j + 2] = 2

    matrix_two[n - 2][2] = 2
    matrix_two[n - 2][0] = 2
    matrix_two[n - 1][3] = 2
    matrix_two[n - 1][1] = 2

    i = i + 1
    if i >= n:
        break

# Third interaction
i = 0
for j in range(0, m):
    if i == j:
        matrix_three[i][j] = 2

    i = i + 1
    if i >= n:
        break

for x in range(0, n):
    for y in range(0, m):
        if matrix_one[x][y] == 2:
            matrix_one[x][y] = spin_z
        else:
            matrix_one[x][y] = identity

        if matrix_two[x][y] == 2:
            matrix_two[x][y] = spin_z
        else:
            matrix_two[x][y] = identity

        if matrix_three[x][y] == 2:
            matrix_three[x][y] = spin_x
        else:
            matrix_three[x][y] = identity

print(matrix_one)
print(matrix_two)
print(matrix_three)
